src/common.rs

```
@@ -10,6 +10,450 @@ pub type TxIndex = U64;
 pub type UncleIndex = U64;
 pub type UncleCount = U64;
 
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(remote = "Self")]
+#[serde(rename_all = "camelCase")]
+pub enum DefaultBlock {
+    Earliest,
+    Latest,
+    Pending,
+    Number(U64),
+}
+
+//impl<'de> EnumAccess<'de> for DefaultBlock {}
+
+impl<'de> Deserialize<'de> for DefaultBlock {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        Self::deserialize(UnitVariantDeserializer(deserializer))
+    }
+}
+struct UnitVariantDeserializer<D>(D);
+
+impl<'de, D> de::Deserializer<'de> for UnitVariantDeserializer<D>
+where
+    D: serde::Deserializer<'de>,
+{
+    type Error = D::Error;
+
+    fn deserialize_any<V>(self, _visitor: V) -> Result<V::Value, D::Error>
+    where
+        V: Visitor<'de>,
+    {
+        todo!()
```

This will crash on invalid input.

---

Yes, is it really sensible to keep implementing this? Please refer to the comment in this PR which I have tagged you in.

---

Under no circumstances should an invalid JSONRPC request kill the entire node / RPC daemon. This would instantly become a form of DoS, especially since `eth_` namespace requests are often Internet-facing.

---

If you can't implement it at the moment, you could return an error to the client saying something like 501 Not Implemented or some other graceful return. `todo!` is going to crash Rust :)

---

Ok, thank you. Can we allow DefaultBlock to be a type alias for String? And leave validation ("earliest"/"latest"/"pending"/hexstring) to each implementation of EthApi trait?

---

Nvm, I solved it inspired by Gelfand's solution for returning either a full tx object or partial (H256) :D